metering/logging.go

```diff
@@ -0,0 +1,81 @@
+package metering
+
+import (
+	"bufio"
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var encoder *json.Encoder
+var meteringBuffer *bufio.Writer
+var logger *logrus.Entry
+var writelock sync.Mutex
+
+func init() {
+	initFromEnv()
+}
+
+func initFromEnv() {
+	fname := os.Getenv("METERING_FILENAME")
+	if fname != "" {
+		err := SetMeteringFile(fname)
+		if err != nil {
+			panic(errors.Wrapf(err, "Failed to open file '%s' which was specified by the env var 'METERING_FILENAME'", fname))
+		}
+	} else {
+		meteringBuffer = bufio.NewWriter(os.Stdout)
+		encoder = json.NewEncoder(meteringBuffer)
```

Comment:
Why aren't we using logrus with a JSONFormatter?

Reply:
There is no way to flush logrus I believe.

Reply:
I guess I could set logrus to use a buffered output and then flush it manually (like this).

Reply:
And logrus has a "msg" field by default - which we don't need